src/orion/storage/backup.py

```diff
@@ -0,0 +1,734 @@
+
+logger = logging.getLogger(__name__)
+
+COL_EXPERIMENTS = "experiments"
+COL_ALGOS = "algo"
+COL_BENCHMARKS = "benchmarks"
+COL_TRIALS = "trials"
+
+COLLECTIONS = {"experiments", "algo", "benchmarks", "trials"}
+EXPERIMENT_RELATED_COLLECTIONS = {"algo", "trials"}
+
+STEP_COLLECT_EXPERIMENTS = 0
+STEP_CHECK_BENCHMARKS = 1
+STEP_CHECK_DST_EXPERIMENTS = 2
+STEP_CHECK_SRC_EXPERIMENTS = 3
+STEP_DELETE_OLD_DATA = 4
+STEP_INSERT_NEW_DATA = 5
+STEP_NAMES = [
+    "Collect source experiments to load",
+    "Check benchmarks",
+    "Check destination experiments",
+    "Check source experiments",
+    "Delete data to replace in destination",
+    "Insert new data in destination",
+]
+
+
+def dump_database(storage, dump_host, name=None, version=None, overwrite=False):
+    """Dump a database
+
+    Parameters
+    ----------
+    storage: BaseStorageProtocol
+        storage of database to dump
+    dump_host:
+        file path to dump into (dumped file will be a pickled file)
+    name:
+        (optional) name of experiment to dump (by default, full database is dumped)
+    version:
+        (optional) version of experiment to dump.
+        By default, use the latest version of provided `name`.
+    overwrite:
+        (optional) define how to manage destination file if already exists.
+        If false (default), raise an exception.
+        If true, delete existing file and create a new one with dumped data.
+    """
+    dump_host = os.path.abspath(dump_host)
+
+    # For pickled databases, make sure src is not dst
+    if hasattr(storage, "_db"):
+        orig_db = storage._db
+        if isinstance(orig_db, PickledDB) and dump_host == os.path.abspath(
+            orig_db.host
+        ):
+            raise DatabaseError("Cannot dump pickleddb to itself.")
+
+    if os.path.exists(dump_host):
+        if overwrite:
+            # NB: The existing file may have been created to lock file name,
+            # for e.g. if called from Web API, to avoid different API calls
+            # using same file name. In such case, deleting the file may allow
+            # another web API call to lock file name just before the file
+            # is recreated. To prevent this, it's better to just erase file
+            # content, instead of delete/recreate it.
+            with open(dump_host, "wb"):
+                pass
+            assert os.path.exists(dump_host)
+            assert os.stat(dump_host).st_size == 0
+            logger.info(f"Overwriting previous output at {dump_host}")
+        else:
+            raise DatabaseError(
+                f"Export output already exists (specify `--force` to overwrite) at {dump_host}"
+            )
+
+    try:
+        dst_storage = setup_storage(
+            {"database": {"host": dump_host, "type": "pickleddb"}}
+        )
+        logger.info(f"Dump to {dump_host}")
+        _dump(storage, dst_storage, name, version)
+    except Exception as exc:
```

Comment:
If the path already exists and --force is used, this would have the effect of deleting the target db. I think the safest way would be to work on a tmp file-path and then move it to the target db path only if the process was successful. This case should be tested as well (db exists and --force is used but the process crash, then the db should have been left intact).

Reply:
Done ! https://github.com/Epistimio/orion/pull/1082/commits/922744d70f10152a03a8a78cacee04421d260ec2

All comments addressed!